jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -355,6 +355,155 @@ ecma_builtin_typedarray_prototype_map (ecma_value_t this_arg, /**< this argument
 } /* ecma_builtin_typedarray_prototype_map */
 
 /**
+ * The routine reduce and reduceRight share the similar structure.
+ * And we use 'is_right' to distinguish them
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_typedarray_prototype_reduce_with_direction (ecma_value_t this_arg, /**< this argument */
+                                                         ecma_value_t cb_func_val, /**< callback function */
+                                                         ecma_value_t initial_val, /**< initial value */
+                                                         bool is_right) /**< choose order, true is reduceRight */
+{
+  if (!ecma_is_typedarray (this_arg))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Argument 'this' is not a TypedArray."));
+  }
+
+  if (!ecma_op_is_callable (cb_func_val))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Callback function is not callable."));
+  }
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (this_arg);
+  uint32_t len = ecma_typedarray_get_length (obj_p);
+
+  if (len == 0 && ecma_is_value_undefined (initial_val))
+  {
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Initial value cannot be undefined."));
+  }
+
+  ecma_value_t accumulator = ecma_make_simple_value (ECMA_SIMPLE_VALUE_UNDEFINED);
+  int64_t index = is_right ? (len - 1) : 0;
```

Comment:
int64? As far as I know we cannot allocate such amount of memory. And int64 is not a native type on 32 bit machines.